src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/InterchangeHeader.java

```diff
@@ -66,30 +59,48 @@ public String getKey() {
     }
 
     @Override
-    public String getValue() {
-        String timestamp = DATE_FORMAT.format(translationTime);
-        String formattedSequenceNumber = String.format("%08d", sequenceNumber);
-        return "UNOA:2" + "+" + sender + "+" + recipient + "+" + timestamp + "+" + formattedSequenceNumber;
-    }
-
-    @Override
-    protected void validateStateful() throws EdifactValidationException {
-        if (sequenceNumber == null) {
+    public void validate() throws EdifactValidationException {
+        if (StringUtils.isBlank(getSender())) {
+            throw new EdifactValidationException(getKey() + ": Attribute sender is required");
+        }
+        if (StringUtils.isBlank(getRecipient())) {
+            throw new EdifactValidationException(getKey() + ": Attribute recipient is required");
+        }
+        if (getTranslationTime() == null) {
+            throw new EdifactValidationException(getKey() + ": Attribute translationTime is required");
+        }
+        if (getSequenceNumber() == null) {
             throw new EdifactValidationException(getKey() + ": Attribute sequenceNumber is required");
         }
-        if (sequenceNumber < 1 || sequenceNumber > MAX_INTERCHANGE_SEQUENCE) {
+        if (getSequenceNumber() < 1 || getSequenceNumber() > MAX_INTERCHANGE_SEQUENCE) {
             throw new EdifactValidationException(
                 getKey() + ": Attribute sequenceNumber must be between 1 and " + MAX_INTERCHANGE_SEQUENCE);
         }
     }
 
-    @Override
-    public void preValidate() throws EdifactValidationException {
-        if (sender.isEmpty()) {
-            throw new EdifactValidationException(getKey() + ": Attribute sender is required");
+    private static Long parseLong(String longString) {
+        if (StringUtils.isBlank(longString)) {
+            return null;
         }
-        if (recipient.isEmpty()) {
-            throw new EdifactValidationException(getKey() + ": Attribute recipient is required");
+        try {
+            return Long.parseLong(longString);
+        } catch (Exception ex) {
+            throw new EdifactValidationException(KEY + ": Error parsing long value", ex);
+        }
+    }
+
+    public static Instant getTranslationTime(String translationTimeStr) {
+        if (StringUtils.isBlank(translationTimeStr)) {
+            return null;
+        }
+        ZonedDateTime translationTime;
+        try {
+            translationTime = ZonedDateTime.parse(
+                translationTimeStr,
+                DateTimeFormatter.ofPattern("yyMMdd:HHmm").withZone(TimestampService.UK_ZONE));
+        } catch (Exception ex) {
+            throw new EdifactValidationException(KEY + ": Error parsing time", ex);
         }
+        return translationTime.toInstant();
```

Comment:
Code style preference difference here. Variables declared without instantiation looks worse to me than not having returns as the last statement.

```suggestion
        try {
            ZonedDateTime translationTime = ZonedDateTime.parse(
                translationTimeStr,
                DateTimeFormatter.ofPattern("yyMMdd:HHmm").withZone(TimestampService.UK_ZONE));
            return translationTime.toInstant();
        } catch (Exception ex) {
            throw new EdifactValidationException(KEY + ": Error parsing time", ex);
        }
```

Reply:
I wanted to keep only the potentially dangerous code inside the try-catch block but I agree that uninitialized variables don't look good.